src/script/script.cpp

```diff
@@ -255,6 +256,60 @@ bool CScript::IsPayToScriptHash() const
             (*this)[22] == OP_EQUAL);
 }
 
+bool CScript::IsRedeemScriptReveal(CScript scriptpubkey) const{
+    CScript check_spk, redeemScript = scriptpubkey;
+    if (
+        // these magic numbers correspond to a typical(as of May 2021) atomicdex HLTC
+        redeemScript[0] == OP_RETURN &&
+        redeemScript[3] == OP_IF &&
+        redeemScript[9] == OP_NOP2 &&
+        redeemScript[10] == OP_DROP &&
+        redeemScript[45] == OP_CHECKSIG &&
+        redeemScript[46] == OP_ELSE &&
+        redeemScript[47] == OP_SIZE &&
+        redeemScript[48] == 0x01 &&
+        redeemScript[49] == 0x20 &&
+        redeemScript[50] == OP_EQUALVERIFY &&
+        redeemScript[51] == OP_HASH160 &&
+        redeemScript[73] == OP_EQUALVERIFY &&
+        redeemScript[108] == OP_CHECKSIG &&
+        redeemScript[109] == OP_ENDIF &&
+        redeemScript.size() == 110
```

Comment:
I'm unsure in regards to the behavior in case of getting the out-of-bounds index, but I suppose nothing good will happen. The size check should be performed before getting any element by index.